test/extended/syncer/syncer_utils.go

```diff
@@ -0,0 +1,142 @@
+type SyncTargetOption func(*SyncTarget)
+
+// SetSyncTargetName sets the SyncTarget's name
+func SetSyncTargetName(name string) SyncTargetOption {
+	return func(s *SyncTarget) {
+		s.Name = name
+	}
+}
+
+// SetSyncTargetOutputFilePath sets the SyncTarget's OutputFilePath
+func SetSyncTargetOutputFilePath(path string) SyncTargetOption {
+	return func(s *SyncTarget) {
+		s.OutputFilePath = path
+	}
+}
+
+// NewSyncTarget create a new customized SyncTarget
+func NewSyncTarget(opts ...SyncTargetOption) SyncTarget {
+	defaultSyncTarget := SyncTarget{
+		Name:            "mysyncer-" + exutil.GetRandomString(),
+		WorkSpaceServer: "",
+	}
+	for _, o := range opts {
+		o(&defaultSyncTarget)
+	}
+	return defaultSyncTarget
+}
+
+// Create SyncTarget
+func (s *SyncTarget) Create(k *exutil.CLI) {
+	if s.WorkSpaceServer == "" {
+		s.WorkSpaceServer = k.WorkSpace().ServerURL
+	}
+	s.CreateAsExpectedResult(k, true, "Wrote physical cluster manifest to")
+
+}
+
+// CreateAsExpectedResult creates SyncTarget CR and checks the created result is as expected
+func (s *SyncTarget) CreateAsExpectedResult(k *exutil.CLI, successFlag bool, containsMsg string) {
+	kcpServerVersion, getError := exutil.GetKcpServerVersion(k)
+	o.Expect(getError).ShouldNot(o.HaveOccurred())
+	o.Expect(kcpServerVersion).Should(o.MatchRegexp(`v\d+(.\d+){0,2}`))
+	msg, createError := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("kcp").Args("--server="+s.WorkSpaceServer, "workload", "sync", s.Name,
+		"--syncer-image=ghcr.io/kcp-dev/kcp/syncer:"+kcpServerVersion, "--output-file="+s.OutputFilePath).Output()
+	if successFlag {
+		o.Expect(createError).ShouldNot(o.HaveOccurred())
+		o.Expect(msg).Should(o.ContainSubstring(containsMsg))
+	} else {
+		o.Expect(createError).Should(o.HaveOccurred())
+		o.Expect(fmt.Sprint(msg)).Should(o.ContainSubstring(containsMsg))
+	}
+}
+
+// GetFieldByJSONPath gets specific field value of the SyncTarget by jsonpath
+func (s *SyncTarget) GetFieldByJSONPath(k *exutil.CLI, JSONPath string) (string, error) {
+	return k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+s.WorkSpaceServer, "syncTarget", s.Name, "-o", "jsonpath="+JSONPath).Output()
+}
+
+// CheckReady checks whether the SyncTarget is ready
+func (s *SyncTarget) CheckReady(k *exutil.CLI) (bool, error) {
+	readyFlag, err := s.GetFieldByJSONPath(k, `{.status.conditions[?(@.type=="Ready")].status}`)
+	if err != nil {
+		e2e.Logf(`Getting SyncTarget/%s status failed: "%v"`, s.Name, err)
+		return false, err
+	}
+	e2e.Logf("SyncTarget/%s ready condition is %s", s.Name, readyFlag)
+	return strings.EqualFold(readyFlag, "True"), nil
+}
+
+// WaitUntilReady waits the SyncTarget become ready
+func (s *SyncTarget) WaitUntilReady(k *exutil.CLI) {
+	err := wait.Poll(5*time.Second, 180*time.Second, func() (bool, error) {
+		SyncTargetReady, err := s.CheckReady(k)
+		if err != nil {
+			return SyncTargetReady, err
+		}
+		return SyncTargetReady, nil
```

Comment:
Line 97 ~ line 100 can merge into one line:
```
                return SyncTargetReady, err
```
If so, line 96 ~ line 100 can further merge into one line:
```suggestion
                return s.CheckReady(k)
```

Reply:
Good point! As the checkReady alrady print the err log, we could merge the conditions. Addresses.